File: backend/src/main/java/com/jobtracker/entity/Job.java
package com.jobtracker.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})  // 忽略懒加载相关属性
@Table(uniqueConstraints = {
    @UniqueConstraint(columnNames = {"title", "company"})
})
public class Job {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String title;
    private String company;
    private String location;
    private String url;
    private String source;
    @Column(columnDefinition = "TEXT")
    private String description;
}

File: frontend/src/types/Job.ts
export interface Job {
  id: number;           // 对应后端id (Long)
  title: string;        // 对应后端title
  company: string;      // 对应后端company
  location: string;     // 对应后端location
  source: string;       // 对应后端source
  url?: string;         // 对应后端url
  description?: string; // 对应后端description
  isFavorite?: boolean; // 是否被当前用户收藏
  matchScore?: number;  // 匹配分数 (仅用于简历推荐结果)
}
